### ISSUE TITLE
[Feature] Follow XDG directory specifications

### ISSUE THREAD

**Issue by JediMaster25** [stashapp/stash#2814]
There was [an issue](https://github.com/stashapp/stash/issues/287) opened for this already, and it was closed to

> bring the number of open issues down and close this.

The result is that stash still doesn't follow the xdg specifications three years later, thanks to the brilliance of this guy.

If the xdg folders exist and are in the environment, then the config should go to `$XDG_CONFIG_HOME/stash`, the database should go to `$XDG_DATA_HOME/stash`, and any non-essential data should go to `$XDG_CACHE_HOME/stash`.

**Comment by scruffynerf** [comment]
Agree, but with the caveat that that's only true on first run.  Once it's got a config, that matters more than any xdg folders in the environment.  In other words, it should only use those environments to OFFER to configure itself the first time Stash is run, by default.  Right now, we literally ask new users what to do with no real 'newbie' idea of what they are being asked.  Offering this as an option makes sense.  I'm not sure putting thing in 3 places is a great idea, but it's an option.

Suggested UI change for going forward was proposed here:
https://discord.com/channels/559159668438728723/999385036115685477/1000203010221281340

**Comment by aghoulcoder** [comment]
@JediMaster25 is absolutely correct.
Also using XDG paths by default is related to the current bug with paths being relative to the cwd of the executable instead of relative to the config file, as is stated in the initial setup. #3029
